conjure-go-client/httpclient/client.go

```diff
@@ -154,26 +163,19 @@ func (c *clientImpl) doOnce(
 
 	// 2. create the transport and client
 	// shallow copy so we can overwrite the Transport with a wrapped one.
-	clientCopy := c.client
+	clientCopy := *c.client.CurrentHTTPClient()
 	transport := clientCopy.Transport // start with the concrete http.Transport from the client
```

Comment:
We should update this comment since it's no longer true, especially since we've moved the client middleware (metrics/trace) over to the httpclient builder and the ordering is very important